### PR TITLE
Bump MSRV to 1.63

### PR DESCRIPTION
**Pull request by Kixunil**
The version 1.63 satisfies our requirements for MSRV and provides significant benefits so this commit bumps it. This commit also starts using weak dependencies.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -7,10 +7,10 @@
 //!
 //! To minimize dependencies, some functions are feature-gated. To generate
 //! random keys or to re-randomize a context object, compile with the
-//! `rand-std` feature. If you are willing to use the `rand-std` feature, we
-//! have enabled an additional defense-in-depth sidechannel protection for
-//! our context objects, which re-blinds certain operations on secret key
-//! data. To de/serialize objects with serde, compile with "serde".
+//! `rand` and `std` features. If you are willing to use the these feature,
```
Comment (tcharding) [inline src/lib.rs:10]:
```suggestion
//! `rand` and `std` features. If you are willing to use these feature,
```

**Review by tcharding — APPROVED** [review]
Sans the typo, ACK 99b2cdc03dad83cf6dc9bf31472323edd563453d

**Review by tcharding — APPROVED** [review]
ACK 55c2efc3203417be91c6d1db9258b430e4fe8b82

**Review by apoelstra — APPROVED** [review]
ACK 55c2efc3203417be91c6d1db9258b430e4fe8b82

**Pull request merged**